portal/src/main/webapp/js/src/civic/civicservice.js

```diff
@@ -0,0 +1,164 @@
+                if (ids.length > 0) {
+                    ids += ',';
+                }
+                ids += geneSymbol;
+
+                // To prevent the request from growing too large, we send it off
+                // when it reaches this limit and start a new one
+                if (ids.length >= 1900) {
+                    promises.push(getCivicGenesBatch(ids));
+                    ids = '';
+                }
+            });
+            if (ids.length > 0) {
+                promises.push(getCivicGenesBatch(ids));
+            }
+
+            // We're explicitly waiting for all promises to finish (done or fail).
+            // We are wrapping them in another promise separately, to make sure we also 
+            // wait in case one of the promises fails and the other is still busy.
+            var mainPromise = $.when.apply($, $.map(promises, function(promise) {
+                var wrappingDeferred = $.Deferred();
+                promise.always(function (result) {
+                    wrappingDeferred.resolve();
+                });
+                return wrappingDeferred.promise();
+            }));
+
+            return mainPromise.then(function() {
+                return civicGenes;
+            });
+        },
+
+        getMatchingCivicVariants: function(civicVariants, proteinChange) {
+            var matchingCivicVariants = [];
+            
+            // If present, add the exact match first
+            var civicVariant = civicVariants[proteinChange];
+            if (typeof civicVariant !== 'undefined') {
+                matchingCivicVariants.push(civicVariant);
+            }
+
+            // Match any other variants after splitting the name on + or /
+            $.each(civicVariants, function(name, civicVariant) {
+                var split = name.split(/[+\/]/);
+                if (split.length > 1 && split.indexOf(proteinChange) >= 0) {
+                    matchingCivicVariants.push(civicVariant);
+                }
+            });
+            
+            return matchingCivicVariants;
+        },
+        
+        getCivicVariant: function(civicVariant) {
+            var deferred = $.Deferred();
+
+            if (civicVariant.hasOwnProperty('description')) {
+                // Variant info has already been loaded
+                deferred.resolve();
+            }
+            else {
+                $.ajax({
+                        type: 'GET',
+                        url: civicUrl + 'variants/' + civicVariant.id,
+                        dataType: 'json'
+                    })
+                    .done(function (result) {
+                        if (_.isString(result)) {
+                            result = $.parseJSON(result);
+                        }
+                        civicVariant.description = result.description;
+
+                        civicVariant.url = 'https://civic.genome.wustl.edu/#/events/genes/' + civicVariant.geneId +
```

Comment:
This is the link to the website entry, which is (at least now) not configurable, as opposed to the API url.